pico/types/src/main/java/io/helidon/pico/types/AnnotationAndValue.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.types;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents an annotation along with its value(s).
+ */
+public interface AnnotationAndValue {
+
+    /**
+     * The type name, e.g., {@link jakarta.inject.Named} -> "jakarta.inject.Named".
+     *
+     * @return the annotation type name
+     */
+    TypeName getTypeName();
+
+    /**
+     * The value.
+     *
+     * @return The string value of (all attributes of) the annotation (expressed as toString()), or null if no value is present
+     */
+    String getValue();
+
+    /**
+     * Get a value of an annotation property.
+     *
+     * @param name name of the annotation property
+     * @return string value of the property
+     */
+    String getValue(String name);
+
+    /**
+     * Get a key-value of all the annotation properties.
+     *
+     * @return key-value pairs of all the properties present
+     */
+    Map<String, String> getValues();
+
+    /**
+     * Determines whether the value is set to a non-null and non-blank value.
+     *
+     * @return true if the value provided is non-null and non-blank.
+     */
+    default boolean hasValue() {
+        return hasValue(getValue());
+    }
+
+    /**
+     * Helper method to determine if the value is present (i.e., non-null and non-blank).
+     *
+     * @return true if the value provided is non-null and non-blank.
+     */
+    static boolean hasValue(String val) {
```

Comment:
For optional string attributes on annotations, most people as a convention will use 'default ""'. That does not technically count as having a value.  There are also many places in code coming that will need the "not null and not blank" API like this.  We can think about renaming it, but taking it out is not an option.

Reply:
If you look at our own annotations, we do use empty strings a lot of time (when feasible), but sometimes we had to implement a custom default value, to allow usage of empty string as a valid value.
So yes, in a lot of cases empty string means not having a value. But is some cases it does not.

Reply:
First, I need a method that will do this check in its implementation somewhere in the API. If not here then I'll need to pick a new spot for that impl. This avoids having to write "val != null && !val.isBlank()" in many, many places that will soon be coming.  I can conceptually use "empty" instead of "blank" with no adverse effects, but I thought blank might be better to use in this case. This is point 1.

Second, are you suggesting we need to distinguish *at compile time* a need to differentiate the default value of "" vs intended/ascribed value of "" by the developer? If so then that simply doesn't exist, nor do I believe have any need for it since the processing will behave the same regardless of how it was originated using defaults or via an override.  If it's blank (or empty) it doesn't have a value - plain and simple.

So the realistic options I see are:
0) Use javadoc to clearly describe how "" is considered to not have any value from our opinionated PoV on this class at leasy.
1) use "empty" instead of "blank" for hasValue()
2) use a better description: hasValue() -> hasNonBlankValue() and/or hasNonEmptyValue()